Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h

```diff
@@ -413,6 +418,58 @@ class CEXIETHERNET : public IEXIDevice
 #endif
   };
 
+  class BuiltInBBAInterface : public NetworkInterface
+  {
+  public:
+    BuiltInBBAInterface(CEXIETHERNET* eth_ref, std::string dns_ip)
+        : NetworkInterface(eth_ref), m_dns_ip(std::move(dns_ip))
+    {
+    }
+
+  public:
+    bool Activate() override;
+    void Deactivate() override;
+    bool IsActivated() override;
+    bool SendFrame(const u8* frame, u32 size) override;
+    bool RecvInit() override;
+    void RecvStart() override;
+    void RecvStop() override;
+
+  private:
+    std::string m_mac_id;
+    std::string m_dns_ip;
+    bool active = false;
+    u16 ip_frame_id = 0;
+    u8 queue_read = 0;
+    u8 queue_write = 0;
+    u16 queue_data_size[16]{};
+    char queue_data[16][2048]{};
+    std::mutex mtx;
+    bool isSent = false;
+#if defined(WIN32) || defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) ||          \
+    defined(__OpenBSD__) || defined(__NetBSD__) || defined(__HAIKU__)
+    StackRef network_ref[10]{};  // max 10 at same time, i think most gc game had a limit of 8 in
+                                 // the gc framework
+    char m_in_frame[9004]{};
```

Comment:
the element type of these arrays should probably be `u8`? might be nice to make them allocated outside the object as well.